samples/client/petstore/typescript-fetch/builds/oneOf/models/TestArrayResponse.ts

```diff
@@ -0,0 +1,89 @@
+/* tslint:disable */
+/* eslint-disable */
+/**
+ * testing oneOf without discriminator
+ * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
+ *
+ * The version of the OpenAPI document: 1.0.0
+ * 
+ *
+ * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
+ * https://openapi-generator.tech
+ * Do not edit the class manually.
+ */
+
+import type { TestA } from './TestA';
+import {
+    instanceOfTestA,
+    TestAFromJSON,
+    TestAFromJSONTyped,
+    TestAToJSON,
+} from './TestA';
+import type { TestB } from './TestB';
+import {
+    instanceOfTestB,
+    TestBFromJSON,
+    TestBFromJSONTyped,
+    TestBToJSON,
+} from './TestB';
+
+/**
+ * @type TestArrayResponse
+ * 
+ * @export
+ */
+export type TestArrayResponse = Array<TestA> | Array<TestB> | Array<string>;
+
+export function TestArrayResponseFromJSON(json: any): TestArrayResponse {
+    return TestArrayResponseFromJSONTyped(json, false);
+}
+
+export function TestArrayResponseFromJSONTyped(json: any, ignoreDiscriminator: boolean): TestArrayResponse {
+    if (json == null) {
+        return json;
+    }
+    if (!(json instanceof Object)){
+        return {} as any;
+    }
+    if (Array.isArray(json) && json.every(item => item instanceof String)) {
+        return json as Array<string>;
+    }
+    if (Array.isArray(json) && json.every(item => item instanceof Object)) {
```

Comment:
same here: `instanceof Object` is probably not what you need.
e.g. `[] instanceof Object // true`
use `typeof` instead